File: Count Nodes Equal to Average of Subtree - Leetcode 2265/Count Nodes Equal to Average of Subtree - Leetcode 2265.java
public class Solution {
    private int numNodes = 0;

    public int averageOfSubtree(TreeNode root) {
        dfs(root);
        return numNodes;
    }

    private int[] dfs(TreeNode root) {
        if (root == null) return new int[] {0, 0}; // {node count, sum of values}

        int[] left = dfs(root.left);
        int[] right = dfs(root.right);

        int nodeCount = 1 + left[0] + right[0];
        int sum = root.val + left[1] + right[1];
        int avg = sum / nodeCount;

        if (root.val == avg) numNodes++;

        return new int[] {nodeCount, sum};
    }
}

File: Count Nodes Equal to Average of Subtree - Leetcode 2265/Count Nodes Equal to Average of Subtree - Leetcode 2265.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = val;
 *     this.left = left;
 *     this.right = right;
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number}
 */
var averageOfSubtree = function(root) {
    let numNodes = 0;

    function dfs(node) {
        if (!node) return [0, 0]; // [node count, sum of values]

        let [leftCount, leftSum] = dfs(node.left);
        let [rightCount, rightSum] = dfs(node.right);

        let nodeCount = 1 + leftCount + rightCount;
        let sum = node.val + leftSum + rightSum;
        let avg = Math.floor(sum / nodeCount);

        if (node.val === avg) numNodes++;

        return [nodeCount, sum];
    }

    dfs(root);
    return numNodes;
};
